src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SEPARATOR = ",";
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] lines = readFromFile(fromFileName).split(System.lineSeparator());
+        int supplyCount = countSpecificType(lines, SUPPLY);
+        int byuCount = countSpecificType(lines, BUY);
+        int resultCount = supplyCount - byuCount;
+        writeToFile(toFileName, toCsvFormat(supplyCount, byuCount, resultCount));
+    }
+
+    private String readFromFile(String fromFileName) {
```

Comment:
The method 'readFromFile' should not be public as it is only used within the same class. Adjust the access modifier to 'private' to follow the encapsulation principle.